tuf/tuf.go

```diff
@@ -245,6 +245,17 @@ func (tr *Repo) GetDelegationRole(name string) (data.DelegationRole, error) {
 				if err != nil {
 					return err
 				}
+				// Check all public key certificates in the role for expiry
+				// Currently we do not reject expired delegation keys but warn if they might expire soon or have already
+				for keyID, pubKey := range delgRole.Keys {
+					certFromKey, err := utils.LoadCertFromPEM(pubKey.Public())
+					if err != nil {
+						continue
+					}
+					if err := utils.ValidateCertificate(certFromKey, true); err != nil {
```

Comment:
This check probably needs to become more explicit as well. Right now it would proceed even if the problem was something like an RSA key being too short (which we should treat as a hard failure). Maybe create an `ErrCertExpired` (if it doesn't exist somewhere already) and only allow continuation if we get that specific error.